src/BluetoothGattService.cpp

```diff
@@ -117,7 +118,9 @@ BluetoothDevice BluetoothGattService::get_device ()
         throw BluetoothException(error_msg);
     }
 
-    return BluetoothDevice(device);
+    auto res = BluetoothDevice(device);
+    g_object_unref(device);
```

Comment:
I'm not sure I understand your comment.
The above code release the device object returned from `device1_proxy_new_for_bus` method and not the object path as returned from `gatt_service1_get_device (object)`
Also doesn't "transfer full" implies it's the caller responsibility to release the object?
https://github.com/intel-iot-devkit/tinyb/blob/master/src/generated-code.c#L5628